hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/DefaultWriteBehindProcessor.java

```diff
@@ -420,8 +424,9 @@ private void sort(List<DelayedEntry> entries) {
 
     private void sleepSeconds(long secs) {
         try {
-            TimeUnit.SECONDS.sleep(secs);
+            SECONDS.sleep(secs);
         } catch (InterruptedException e) {
+            currentThread().interrupt();
             logger.warning(e);
```

Comment:
Do we want to log it? Because you already have handled id (by breaking the loop and restoring the flag).